### DISCUSSION TITLE
core.UDPConn这个结构可不可以用net.PacketConn替代?

### DISCUSSION THREAD

**Discussion by DumAdudus** [apernet/hysteria#405] [General]
https://github.com/HyNetwork/hysteria/blob/aad4c44b3d4f885a118026e692c59110ba709d49/pkg/core/client.go#L322-L326
提出这个问题的原因是Clash.Meta为了支持Hysteria，针对这个结构体做了以下的改动：
https://github.com/MetaCubeX/Clash.Meta/blob/81d5c8b3e8b876817f6bcc3601bf9a5fee3bb90f/transport/hysteria/core/client.go#L325-L333

**Comment by tobyxdd** [comment]
这里面 `SetDeadline`/`SetReadDeadline`/`SetWriteDeadline` 的实现是不对的。

UDP 转发的情况下 stream 本身不用来传输数据，只是保持连接，用 close 作为 UDP session 关闭的信号。把 deadline 设到 stream 上相当于给 UDP session 定了个固定的关闭时间了（因为 `Hold` 里一旦 `Read` 返回错误就会 `c.Close()`），和 `SetDeadline` 应有的行为不符。

**Reply by DumAdudus** [reply]
如果限定了这是链条发起端的行为，是不是是可以接受的了？clash.meta应该是作为终端来使用的
